static/js/copy_and_paste.js

```diff
@@ -48,10 +48,28 @@ function construct_copy_div(div, start_id, end_id) {
     var start_recipient_row = rows.get_message_recipient_row(start_row);
     var start_recipient_row_id = rows.id_for_recipient_row(start_recipient_row);
     var should_include_start_recipient_header = false;
-
+    var row;
+    var recipient_row_id;
+    var message;
+    var message_firstp;
     var last_recipient_row_id = start_recipient_row_id;
-    for (var row = start_row; rows.id(row) <= end_id; row = rows.next_visible(row)) {
-        var recipient_row_id = rows.id_for_recipient_row(rows.get_message_recipient_row(row));
+
+    // Check if all messages are from a single user or not.
+    var first_sender_name;
+    var only_one_user = true;
+    for (row = start_row; rows.id(row) <= end_id; row = rows.next_visible(row)) {
+        message = current_msg_list.get(rows.id(row));
+        if (first_sender_name === undefined) {
+            first_sender_name = message.sender_full_name;
+            continue;
+        }
+        if (first_sender_name !== message.sender_full_name) {
+            only_one_user = false;
+        }
```

Comment:
Can you convert this logic to use `message.sender_id` That's more correct from an API perspective (IDs are for uniqueness, names are for display), and is certainly better for weird cases where you have two users with the "same" full name.

Reply:
> and is certainly better for weird cases where you have two users with the "same" full name.

I didn't think of that. Made this change as you suggested. Thanks!